src/main/java/org/folio/inventory/dataimport/handlers/actions/ReplaceHoldingEventHandler.java

```diff
@@ -0,0 +1,104 @@
+      }
+      HoldingsRecord tmpHoldingsRecord = ObjectMapperTool.getMapper()
+        .readValue(dataImportEventPayload.getContext().get(HOLDINGS.value()), HoldingsRecord.class);
+
+      String holdingId = tmpHoldingsRecord.getId();
+      String hrid = tmpHoldingsRecord.getHrid();
+      String instanceId = tmpHoldingsRecord.getInstanceId();
+      String permanentLocationId = tmpHoldingsRecord.getPermanentLocationId();
+      if (StringUtils.isAnyBlank(hrid, instanceId, permanentLocationId, holdingId)) {
+        throw new EventProcessingException(EMPTY_REQUIRED_FIELDS_ERROR_MESSAGE);
+      }
+      prepareEvent(dataImportEventPayload);
+
+      MappingManager.map(dataImportEventPayload);
+
+      Context context = constructContext(dataImportEventPayload.getTenant(), dataImportEventPayload.getToken(), dataImportEventPayload.getOkapiUrl());
+      HoldingsRecordCollection holdingsRecords = storage.getHoldingsRecordCollection(context);
+      HoldingsRecord holding = ObjectMapperTool.getMapper().readValue(dataImportEventPayload.getContext().get(HOLDINGS.value()), HoldingsRecord.class);
+      holding.setId(holdingId);
+      holding.setHrid(hrid);
+      holding.setInstanceId(instanceId);
+      holding.setPermanentLocationId(permanentLocationId);
+
+      holdingsRecords.update(holding, holdingSuccess -> constructDataImportEventPayload(future, dataImportEventPayload, holding),
+        failure -> {
+          LOGGER.error(REPLACE_HOLDING_ERROR_MESSAGE);
+          future.completeExceptionally(new EventProcessingException(REPLACE_HOLDING_ERROR_MESSAGE));
+        });
+    } catch (Exception e) {
+      LOGGER.error(e);
+      future.completeExceptionally(e);
+    }
+    return future;
+  }
+
+  @Override
+  public boolean isEligible(DataImportEventPayload dataImportEventPayload) {
+    if (dataImportEventPayload.getCurrentNode() != null && ACTION_PROFILE == dataImportEventPayload.getCurrentNode().getContentType()) {
+      ActionProfile actionProfile = JsonObject.mapFrom(dataImportEventPayload.getCurrentNode().getContent()).mapTo(ActionProfile.class);
+      return actionProfile.getAction() == REPLACE && actionProfile.getFolioRecord() == HOLDINGS;
+    }
+    return false;
+  }
+
+  private void prepareEvent(DataImportEventPayload dataImportEventPayload) {
+    dataImportEventPayload.getEventsChain().add(dataImportEventPayload.getEventType());
+    dataImportEventPayload.getContext().put(HOLDINGS.value(), new JsonObject().encode());
+    dataImportEventPayload.setCurrentNode(dataImportEventPayload.getCurrentNode().getChildSnapshotWrappers().get(0));
+  }
+
+  private void constructDataImportEventPayload(CompletableFuture<DataImportEventPayload> future, DataImportEventPayload dataImportEventPayload, HoldingsRecord holding) {
+    dataImportEventPayload.getContext().put(HOLDINGS.value(), Json.encodePrettily(holding));
+    dataImportEventPayload.setEventType(DI_INVENTORY_HOLDING_CREATED.value()); //TODO: change on REPLACED
```

Comment:
need to change?

Reply:
@OleksiiKuzminov Yes. It will be changed on "DI_INVENTORY_HOLDING_REPLACED". I added "TODO"s in these places (as I mentioned below). After merging PR in data-import-processing-core I will change it.